pkg/operator/controllers/cloudproviderconfig/cloudproviderconfig_controller.go

```
@@ -0,0 +1,100 @@
+
+// CloudProviderConfigReconciler reconciles the openshift-config/cloud-provider-config ConfigMap
+type CloudProviderConfigReconciler struct {
+	base.AROController
+}
+
+func NewReconciler(Log *logrus.Entry, client client.Client) *CloudProviderConfigReconciler {
+	return &CloudProviderConfigReconciler{
+		AROController: base.AROController{
+			Log:    Log,
+			Client: client,
+			Name:   ControllerName,
+		},
+	}
+}
+
+// Reconcile makes sure that the cloud-provider-config is healthy
+func (r *CloudProviderConfigReconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	instance := &arov1alpha1.Cluster{}
+	err := r.Client.Get(ctx, types.NamespacedName{Name: arov1alpha1.SingletonClusterName}, instance)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if !instance.Spec.OperatorFlags.GetSimpleBoolean(controllerEnabled) {
+		r.Log.Debug("controller is disabled")
+		return reconcile.Result{}, nil
+	}
+
+	r.Log.Debug("running")
+	return reconcile.Result{}, r.updateCloudProviderConfig(ctx)
+}
+
+// SetupWithManager setup our manager
+func (r *CloudProviderConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	cloudProviderConfigPredicate := predicate.NewPredicateFuncs(func(o client.Object) bool {
+		return o.GetName() == cloudProviderConfigName.Name && o.GetNamespace() == cloudProviderConfigName.Namespace
+	})
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.ConfigMap{}, builder.WithPredicates(cloudProviderConfigPredicate)).
+		Named(ControllerName).
+		Complete(r)
+}
+
+func (r *CloudProviderConfigReconciler) updateCloudProviderConfig(ctx context.Context) error {
+	cm := &corev1.ConfigMap{}
+	err := r.Client.Get(ctx, cloudProviderConfigName, cm)
+	if err != nil {
+		return fmt.Errorf("the ConfigMap openshift-config/cloud-provider-config is missing")
+	}
+
+	jsonConfig, ok := cm.Data["config"]
+	if !ok {
+		return fmt.Errorf("field config in ConfigMap openshift-config/cloud-provider-config is missing")
```

Again, I think I'd rather not mask errors completely, can we append the original error to the new one?